src/server/game/AI/SmartScripts/SmartAI.h

```diff
@@ -274,6 +274,12 @@ class SmartGameObjectAI : public GameObjectAI
     void EventInform(uint32 eventId) override;
     void SpellHit(Unit* unit, SpellInfo const* spellInfo) override;
 
+    // Called when the gameobject summon successfully other creature
+    void JustSummoned(Creature* creature) override;
+
+    // Called when a summoned creature dissapears (UnSummoned)
+    void SummonedCreatureDespawn(Creature* unit) override;
```

Comment:
why use unit for the pointer and not creature?

Reply:
Because it's a cherry-pick: https://github.com/TrinityCore/TrinityCore/commit/507a56d62764f63113d35dfcca9273b8d0b16cc5